aws/cognito/lambda/cognito_email_sender/cognito_email_sender.js

```diff
@@ -0,0 +1,81 @@
+const encryptionSDK = require("@aws-crypto/client-node");
+const {NotifyClient} = require("notifications-node-client");
+
+const KEY_ARN = process.env.KEY_ARN;
+const KEY_ALIAS = process.env.KEY_ALIAS;
+const TEMPLATE_ID = process.env.TEMPLATE_ID;
+const NOTIFY_API_KEY = process.env.NOTIFY_API_KEY;
+
+
+exports.handler = async (event) => {
+  // setup the encryptionSDK's key ring
+  const {decrypt} = encryptionSDK.buildDecrypt(encryptionSDK.CommitmentPolicy.FORBID_ENCRYPT_ALLOW_DECRYPT);
+  const generatorKeyId = KEY_ALIAS;
+  const keyIds = [KEY_ARN];
+  const keyring = new encryptionSDK.KmsKeyringNode({generatorKeyId, keyIds});
+
+  // setup the notify client
+  const notify = new NotifyClient("https://api.notification.canada.ca", NOTIFY_API_KEY);
+
+  // decrypt the code to plain text if it exists
+  let plainTextCode;
+  if (event.request.code) {
+    try {
+      // create Buffer from base64 text
+      const codeBuffer = Buffer.from(event.request.code, "base64");
+      // decrypt the code into plaintext using the sdk and keyring
+      const {plaintext} = await decrypt(keyring, Uint8Array.from(codeBuffer));
+      plainTextCode = plaintext;
+    } catch (err) {
+      console.error(
+        `{"status": "failed", "message": "Failed to Decrypt Cognito Code", "error":${JSON.stringify(err)}}`
+      );
+      throw new Error("Could not decrypt Cognito Code");
+    }
+  }
+
+  const userEmail = event.request.userAttributes.email;
+  if(plainTextCode && userEmail){
+    // attempt to send the code to the user through Notify
+    try {
```

Comment:
What do you think about the following refactoring:

````
await notify.sendEmail(TEMPLATE_ID, userEmail, {
  personalisation: {
    passwordReset: event.triggerSource === "CustomEmailSender_ForgotPassword",
    accountVerification: event.triggerSource === "CustomEmailSender_VerifyUserAttribute",
    resendCode: event.triggerSource === "CustomEmailSender_ResendCode",
    code: plainTextCode
  }
});
````

Reply:
So there are much more `triggerSource` values than what are included in the lambda code. If I were to use this approach then an email would be sent even if the `triggerSource` is not one of `CustomEmailSender_ForgotPassword` `CustomEmailSender_VerifyUserAttribute` `CustomEmailSender_ResendCode`

Reply:
![Screen Shot 2022-12-21 at 3 06 21 PM](https://user-images.githubusercontent.com/26819992/208993763-7022cc96-8803-46b9-a79b-cace90f9292a.png)

Reply:
Ok I see. I should have checked that before but since there was no `default` case I thought there was only 3 possibilities.

Reply:
I think I can use that with an additional if statement let me see

Reply:
Probably something like
```
if (event.triggerSource === "CustomEmailSender_ForgotPassword" || event.triggerSource === "CustomEmailSender_VerifyUserAttribute" | event.triggerSource === "CustomEmailSender_ResendCode") {
```

Reply:
I have used the following 

```ts 
[
      "CustomEmailSender_ForgotPassword",
      "CustomEmailSender_ResendCode",
      "CustomEmailSender_VerifyUserAttribute"
    ].includes(event.triggerSource)
```

let me know what you think

Reply:
I like it :)